clients/admin-ui/src/pages/login.tsx

```diff
@@ -71,7 +82,26 @@ const useLogin = () => {
   };
 
   if (token) {
-    router.push(INDEX_ROUTE);
+    const configWizardFlagIsActive = Flags.some(
+      (flag) => flag.name === "configWizardFlag" && flag.isActive
+    );
+
+    if (configWizardFlagIsActive) {
+      if (systems && systems.length > 0) {
+        const datamapRoute = resolveLink({
+          href: DATAMAP_ROUTE,
+          basePath: "/",
+        });
+
+        return features.plus
+          ? router.push(datamapRoute.href)
+          : router.push(SYSTEM_ROUTE);
+      } 
+        router.push(CONFIG_WIZARD_ROUTE);
+      
+    } else {
+      router.push(INDEX_ROUTE);
+    }
```

Comment:
The logic of this block has gotten pretty complex, and the way that it "returns" is by calling `router.push` which can have unintended consequences. I think this could benefit from a helper function that returns the route that should come next (if any), and then `router.push` is only called in one place.

```js
const getRedirectRoute = (systems, ...etc) => {
  if (!token) {
    return undefined
  }
  const configWizardFlagIsActive = ...;
  if (configWizardFlagIsActive) {
    ...
  }
  return INDEX_ROUTE
}

// In useLogin
const redirectRoute = getRedirectRoute(...)
if (redirectRoute) {
  router.redirect(redirectRoute)
}
// maybe other code doesn't need to run if a redirect is going to occur...
```